packages/core-data/src/fetch/__experimental-fetch-remote-url-data.js

```diff
@@ -38,10 +45,17 @@ const fetchRemoteUrlData = async ( url, options = {} ) => {
 		url: prependHTTP( url ),
 	};
 
-	return apiFetch( {
+	if ( CACHE.has( url ) ) {
+		return CACHE.get( url );
+	}
+
+	const response = apiFetch( {
 		path: addQueryArgs( endpoint, args ),
 		...options,
 	} );
+
+	CACHE.set( url, response );
```

Comment:
If we cache the response **promise**, then it means if we abort(reject) the promise, then the same url will always be rejected and cannot be retried. I think we should only cache it if it's resolved.

1. If there is no cache data for `url1`, then show a loading placeholder and send a request. Once it's resolved, set the cache for `url1` and show the result. If the request is aborted, don't set the cache, do nothing instead.
2. Changing from `url1` to `url2` resets the states and repeat (1) for `url2`.
3. Changing back from `url2` to `url1` will immediately show the cached result from (1) if there's one. If `url1` previously failed, then repeat (1).
4. Aborting should still work since they behaves just like rejected promises.

Reply:
Great points. I'm somewhat rushing this and not thinking through. Let me make some changes.

Reply:
Ok I've pushed a change which I believe should only cache valid responses. I'll update the test instructions.